lib/corefoundation/base.rb

```diff
@@ -33,11 +33,12 @@ def self.typecast(cftyperef)
     def initialize(pointer)
       @ptr = FFI::Pointer.new(pointer)
       ObjectSpace.define_finalizer(self, self.class.finalize(ptr))
+      CF.retain(ptr)
     end
 
     # @param [FFI::Pointer] pointer to the address of the object
     def self.finalize(pointer)
-      proc { CF.release(pointer.address) unless pointer.address.zero? }
+      proc { CF.release(pointer) unless pointer.address.zero? }
```

Comment:
There seems to be regression for this statement, looking at https://github.com/chef/corefoundation/pull/31/commits/edea24218f2f4dc94310123f9a7639c8e20bb8ee

It's confusing to decide which change is correct, for what ruby versions.
So, could you add tests for these?

Reply:
The details on this are in PR description. 

If there's any error releasing the pointer, it will fail with a segfault. This also applies to spec runs too.